src/auth/client.ts

```diff
@@ -26,6 +26,10 @@ export interface AuthenticateServerOptions extends AbortOptions {
   hostname?: string
 }
 
+export interface VerifyPeerOptions {
+  verifyPeer?(peerId: PeerId): Promise<boolean>
```

Comment:
This should accept an optional `AbortSignal` to allow cancelling the promise.  The way we do this everywhere else is to pass an optional `AbortOptions` which lets us add extra options in the future in a backwards-compatible way.

Reply:
This is covered by the RequestInit type. That already takes a signal

Reply:
Oh, sorry. I see what you mean now.